### DISCUSSION TITLE
Is it possible to manualy set the domain of charts?

### DISCUSSION THREAD

**Discussion by TimWillebrands** [microsoft/fluentui#32474] [Q&A]
My team and I have a use case where multiple charts are rendered in a list based on user input. We'd like these charts to share the same domain, regardless of the specific data each one contains. For example, if the user selects data from day 1 to day 10, every chart should display a range from day 1 to day 10, even if the data within some charts only spans from day 3 to 7.

We're wondering if there are any built-in ways to achieve this. Otherwise, were considering manually generating the tickvalues for each chart, but that approach feels somewhat scuffed. And would have us reimplement a lot of default behaviour of the ticks.

**Comment by AtishayMsft** [comment]
This is an interesting scenario. We don't provide support to customize the domain currently.
You could try creating the starting and ending points on the x axis with 0 value on the y axis.
We could provide a better solution if you can tell which chart control are you using.

**Reply by TimWillebrands** [reply]
At this moment we're only using the linechart but I foresee us using basically any _x=time_ based chart type. Like area and vertical-bar charts.

We've explored the 'manually setting the ticks' avenue and I think it might be sufficient for our use case. It's even somewhat nice since you can synchronize the ticks of the charts. One thing that might really help though would be a way of hooking into the default tick-generator so you could do something like:
``` tsx
<LineChart
          data={data}
          ...
          tickValues={values => [...values, new Date()]}
/>
```
